### ISSUE TITLE
Hoobs crashes about two to three times per day.

### ISSUE THREAD

**Issue by Corvetteguy19** [hoobs-org/HOOBS#645]
**Description**
When I go to bed at night everything appears to be running okay.  All of my devices respond properly.  The last few nights though I’ve woken up and the HOOBS service has stopped running.  This happened earlier today and just now.  I’m not sure what’s happening as I have no development experience.  

**Version**
It was the previous version to 3.2.6.  I’m also having the same issue with 3.2.6.

**Did you upgrade**
Please let us know if you upgraded from a previous version.
- [ X] Yes
- [ ] No

**Pervious version**
If you upgraded, please let us know your previous version.

**Did you orginally upgrade to HOOBS 3 from HOOBS 2.1.1?**
Did you perform an upgrade from HOOBS 2.1.1? Some things are different.
- [x] Yes
- [ X] No

**What device are you using?**
Please let us know the device you are running HOOBS on.
- [ ] HOOBS Box
- [ ] HOOBS SD Card
- [ X] Custom Install

**If this is a custom install, what's your operating system?**
Please let us know what operating system and version you are using. Ex. Fedora 30 or macOS Catalina.

Rapsberry Pi 4 Model B
Platform: Linux
Distro: Raspian GNU/Linux
Release: 10
Coden

**List your plugins**
Dummy
Fujitsu
Ring
Harmony

**Post your config**
This can be found in the interface, Configuration -> Advanced or you can get it via SSH `cat ~/.hoobs/etc/config.json`

* Remove any passwords from the config file. This is public.

```

{
    "server": {
        "autostart": 0,
        "port": 80,
        "home_setup_id": "Hidden",
        "polling_seconds": 5,
        "origin": "*"
    },
    "client": {
        "country_code": "US",
        "inactive_logoff": 30,
        "postal_code": "80108",
        "theme": "hoobs-light",
        "temp_units": "fahrenheit",
        "locale": "en",
        "default_route": "status"
    },
    "bridge": {
        "port": Hidden,
        "username": "Hidden",
        "pin": "Hidden",
        "name": "HOOBS"
    },
    "description": "",
    "ports": {},
    "accessories": [
        {
            "accessory": "FGLairThermostat",
            "plugin_map": {
                "plugin_name": "homebridge-fujitsu",
                "index": 0
            },
            "name": "Fujitsu Mini Split Master",
            "username": "Hidden",
            "password": "Hidden",
            "interval": "60",
            "region": "us"
        },
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": true,
            "reverse": false,
            "time": 0,
            "name": "Alarm"
        },
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": false,
            "reverse": false,
            "time": 1800000,
            "name": "30 Minute Sleep Timer Master"
        },
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": false,
            "reverse": false,
            "time": 3600000,
            "name": "60 Minute Sleep Timer Master"
        },
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": false,
            "reverse": false,
            "time": 5400000,
            "name": "90 Minute Sleep Timer Master"
        },
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": false,
            "reverse": false,
            "time": 7200000,
            "name": "120 Minute Sleep Timer Master"
        },
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": false,
            "reverse": false,
            "time": 1800000,
            "name": "30 Minute Sleep Timer Guest"
        },
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": false,
            "reverse": false,
            "time": 3600000,
            "name": "60 Minute Sleep Timer Guest"
        },
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": false,
            "reverse": false,
            "time": 5400000,
            "name": "90 Minute Sleep Timer Guest"
        },
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": false,
            "reverse": false,
            "time": 7200000,
            "name": "120 Minute Sleep Timer Guest"
        }
    ],
    "platforms": [
        {
            "platform": "Ring",
            "hideCameraSirenSwitch": false,
            "refreshToken": "Hidden",
            "plugin_map": {
                "plugin_name": "homebridge-ring"
            },
            "hideUnsupportedServices": false,
            "hideInHomeDoorbellSwitch": false,
            "showPanicButtons": true,
            "hideDoorbellSwitch": false,
            "hideCameraMotionSensor": false,
            "alarmOnEntryDelay": false,
            "hideAlarmSirenSwitch": false,
            "beamDurationSeconds": 120,
            "hideLightGroups": true
        },
        {
            "platform": "HarmonyHubWebSocket",
            "name": "Basement TV",
            "hubName": "Basement Hub",
            "mainActivity": "Watch Apple TV",
            "otherPlatforms": [
                {
                    "name": "Family Room TV",
                    "hubName": "Family Room Hub"
                },
                {
                    "name": "Master Bedroom TV",
                    "hubName": "Master Bedroom Hub"
                },
                {
                    "name": "Guest Bedroom TV",
                    "hubName": "Guest Bedroom Hub"
                }
            ],
            "playPauseBehavior": true
        }
    ]
}

```

**Post your log**
You can get the log from the intercafe. This is in the Log section.

```
5/22/2020, 4:39:20 PM [Basement TV] INFO - starting hub discovery
5/22/2020, 4:39:21 PM Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
    at healthCheck (dgram.js:902:11)
    at Socket.send (dgram.js:627:3)
    at Ping.emit (/home/hoobs/.hoobs/node_modules/harmonyhubjs-discover/lib/ping.js:25:15)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)
5/22/2020, 4:39:21 PM Got SIGTERM, shutting down Bridge...
5/22/2020, 4:39:21 PM [Basement TV] INFO - shutdown
5/22/2020, 4:39:22 PM Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
    at healthCheck (dgram.js:902:11)
    at Socket.send (dgram.js:627:3)
    at Ping.emit (/home/hoobs/.hoobs/node_modules/harmonyhubjs-discover/lib/ping.js:25:15)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)
5/22/2020, 4:39:23 PM Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
    at healthCheck (dgram.js:902:11)
    at Socket.send (dgram.js:627:3)
    at Ping.emit (/home/hoobs/.hoobs/node_modules/harmonyhubjs-discover/lib/ping.js:25:15)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)
5/22/2020, 4:39:24 PM Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
    at healthCheck (dgram.js:902:11)
    at Socket.send (dgram.js:627:3)
    at Ping.emit (/home/hoobs/.hoobs/node_modules/harmonyhubjs-discover/lib/ping.js:25:15)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)

```

**Additional context**
I thought the issue was occurring from the smartthings plugin I had, so I removed that.  I’m still having the same issue though, so it’s not that.  Everything was working fine for the most part since I set things up about a week ago.  As an FYI, I’m really new to all of this, but if you give me steps to follow I can generally figure out what needs to be done.  If I knew what a socket dgram was I’d probably be able to figure this out.  Thanks

**Comment by mkellsy** [comment]
I think this is coming from the Harmony plugin. I know the new SmartThings plugin uses sockets too, so maybe the two plugins were fighting.

I found better luck with the older SmartThings plugin, seems more stable.

**Comment by Corvetteguy19** [comment]
> I think this is coming from the Harmony plugin. I know the new SmartThings plugin uses sockets too, so maybe the two plugins were fighting.
> 
> I found better luck with the older SmartThings plugin, seems more stable.

If I were to restore my configuration do you think that might help or do I need to do a system restore?

**Comment by mkellsy** [comment]
You can try it, but you may need to reset and install the plugins again.

**Comment by Corvetteguy19** [comment]
> You can try it, but you may need to reset and install the plugins again.

Yeah that didn’t work. Sounds like I need to do a reset. What’s the best way to go back to a new version of hoobs and rebuild everything?

**Comment by mkellsy** [comment]
Save your config first. A system backup includes your plugins.

Then there is a reset button on the Help screen.

Install plugins. (Without SmartThings)
And then post your config in the Advanced section of the config screen.

**Comment by Corvetteguy19** [comment]
I went through the reset process, loaded just the harmony plugin and I’m still crashing constantly. One thing I should note. I added another hub and changed the name of one of my existing. I wonder if that’s the problem. At this point I’m ready to wipe my Pi and start with a brand new Hoobs OS or is that what the reset does?

**Comment by mkellsy** [comment]
A reset whipes out all plugins and configs. Essentially the same thing.

**Comment by mkellsy** [comment]
The dual hub thing could be an issue.

**Comment by Corvetteguy19** [comment]
Okay so I did the reset and that didn’t work. I then tried removing hoobs from my pi and reinstalled it and just the harmony plugin. The plugin works fine if I load only one of my hubs. The moment I put more in, that’s when I run into problems. It was working fine for the last week. It was once I updated the plugin that things stopped working. Is there a way to load an old version of the plugin since the update seems to have killed my setup?  Right now I have the new harmony plugin removed.

**Comment by Corvetteguy19** [comment]
Okay I think I might know what’s going on.  I have a mesh network with four hubs running.  From reading other threads, it might have to do with my mesh network.  So my question for you guys is:

1) Is this an issue with the harmony plugin?
2) Is it an issue with Hoobs?
3) Is there something with my mesh network I can do to solve the issue?
4) Who should I reach out to if it’s number 1 & 2?

Thanks guys!

**Comment by askovi** [comment]
Make sure HOOBS is plugged into your main node. That usually fixes a lot of issues.

**Comment by Corvetteguy19** [comment]
> Make sure HOOBS is plugged into your main node. That usually fixes a lot of issues.

I have HOOBS plugged into my main router (xfinity xFi) and I have three xFi pods throughout the house.  My basement hub is connected to the basement xFi pod, my guest bedroom hub is connected to my upstairs xFi pod, and my master bedroom and family room hubs are connected to the main xFi router.  

Is it still possible the mesh network could be causing problems?

**Comment by askovi** [comment]
If the HOOBS bridge was able to connect when you added it to Homekit then it shouldn't be a mesh network problem. Can you post your config with both Harmony Hubs included?

**Comment by Corvetteguy19** [comment]
Here is what my log is showing.  Hopefully there isn’t anything in there that shouldn’t be.  My Config. Is on top.  One thing I noticed.  When I had first installed the plugin and got all of my hubs to load, they automatically showed up in homekit.  Right now, only two of my hubs are showing in homekit.  The basement and Master bedroom hubs.  The other two can be added through the add accessory process.  I’m not sure why that’s happening now, but I feel like it’s related. 

{
    "server": {
        "port": 80,
        "origin": "*",
        "autostart": 0,
        "home_setup_id": "Hidden",
        "polling_seconds": 5
    },
    "client": {
        "default_route": "status",
        "inactive_logoff": 30,
        "theme": "hoobs-light",
        "locale": "en",
        "temp_units": "fahrenheit",
        "country_code": "US",
        "postal_code": "80108"
    },
    "bridge": {
        "name": "HOOBS",
        "port": 51826,
        "pin": "Hidden",
        "username": "Hidden"
    },
    "description": "",
    "ports": {},
    "accessories": [
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": false,
            "reverse": false,
            "time": 1800000,
            "name": "Master Bedroom Sleep Timer 30 Minutes"
        },
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": false,
            "reverse": false,
            "time": 3600000,
            "name": "Master Bedroom Sleep Timer 60 Minutes"
        },
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": false,
            "reverse": false,
            "time": 5400000,
            "name": "Master Bedroom Sleep Timer 90 Minutes"
        },
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": false,
            "reverse": false,
            "time": 7200000,
            "name": "Master Bedroom Sleep Timer 120 Minutes"
        },
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": false,
            "reverse": false,
            "time": 1800000,
            "name": "Guest Bedroom Sleep Timer 30 Minutes"
        },
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": false,
            "reverse": false,
            "time": 3600000,
            "name": "Guest Bedroom Sleep Timer 60 Minutes"
        },
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": false,
            "reverse": false,
            "time": 5400000,
            "name": "Guest Bedroom Sleep Timer 90 Minutes"
        },
        {
            "accessory": "DummySwitch",
            "plugin_map": {
                "plugin_name": "homebridge-dummy",
                "index": 0
            },
            "stateful": false,
            "reverse": false,
            "time": 7200000,
            "name": "Guest Bedroom Sleep Timer 120 Minutes"
        },
        {
            "accessory": "FGLairThermostat",
            "plugin_map": {
                "plugin_name": "homebridge-fujitsu",
                "index": 0
            },
            "name": "Fujitsu Mini Split",
            "username": "pferguson1520@gmail.com",
            "password": "Aviation12",
            "interval": "60",
            "region": "us"
        }
    ],
    "platforms": [
        {
            "platform": "Ring",
            "plugin_map": {
                "plugin_name": "homebridge-ring"
            },
            "refreshToken": "Hidden",
            "hideDoorbellSwitch": false,
            "hideAlarmSirenSwitch": true
        },
        {
            "platform": "SmartThings-v2",
            "plugin_map": {
                "plugin_name": "homebridge-smartthings-v2"
            },
            "name": "SmartThings-v2",
            "direct_port": 8000,
            "temperature_unit": "F",
            "validateTokenId": false,
            "logConfig": {
                "debug": false,
                "showChanges": true,
                "hideTimestamp": true,
                "hideNamePrefix": false,
                "file": {
                    "enabled": true,
                    "level": "Good"
                }
            },
            "app_id": "Hidden",
            "app_url": "https://graph-na04-useast2.api.smartthings.com:443/api/smartapps/installations/",
            "access_token": "Hidden"
        },
        {
            "platform": "HarmonyHubWebSocket",
            "plugin_map": {
                "plugin_name": "homebridge-harmony"
            },
            "otherPlatforms": [
                {
                    "name": "Family Room TV",
                    "hubName": "Family Room Hub"
                },
                {
                    "name": "Master Bedroom TV",
                    "hubName": "Master Bedroom Hub"
                },
                {
                    "name": "Guest Bedroom TV",
                    "hubName": "Guest Bedroom Hub"
                }
            ],
            "name": "Basement TV",
            "hubName": "Basement Hub",
            "mainActivity": "Watch Apple TV"
        }
    ]
}



5/23/2020, 8:35:11 PM [HarmonyHubWebSocket] HarmonyPlatform Init
5/23/2020, 8:35:11 PM [HarmonyHubWebSocket] DidFinishLaunching
5/23/2020, 8:35:11 PM [HarmonyHubWebSocket] (undefined)INFO - Loading activities...
5/23/2020, 8:35:11 PM [HarmonyHubWebSocket] INFO - starting hub discovery
5/23/2020, 8:35:17 PM [HarmonyHubWebSocket] INFO - stopping hub discovery, hubs found : 10.0.0.108|Basement Hub|16323923
5/23/2020, 8:35:17 PM [HarmonyHubWebSocket] (undefined)INFO - received discovered hubs  10.0.0.108|Basement Hub|16323923
5/23/2020, 8:35:18 PM [HarmonyHubWebSocket] (undefined)INFO - Adding Accessory : undefined-TV
5/23/2020, 8:35:18 PM [HarmonyHubWebSocket] (undefined)INFO - configuring Main TV Service
5/23/2020, 8:35:18 PM [HarmonyHubWebSocket] (undefined)INFO - Creating TV Service
5/23/2020, 8:35:18 PM [HarmonyHubWebSocket] (undefined)INFO - Creating Input Service - Listen to Music
5/23/2020, 8:35:18 PM [HarmonyHubWebSocket] (undefined)INFO - Creating Input Service - Watch Apple TV
5/23/2020, 8:35:18 PM [HarmonyHubWebSocket] (undefined)INFO - Creating Input Service - Xbox
5/23/2020, 8:35:18 PM [HarmonyHubWebSocket] (undefined)WARNING - No main Activity that match config file found, default to first one
5/23/2020, 8:35:18 PM [HarmonyHubWebSocket] (undefined)INFO - Configuring Main Activity Listen to Music
5/23/2020, 8:35:18 PM [HarmonyHubWebSocket] (undefined)INFO - Creating TV Speaker Service
5/23/2020, 8:35:18 PM [HarmonyHubWebSocket] (undefined)WARNING - setupFoundAccessories - TV accessory added in your bridge, if another plugin is exposing a TV accessory this one might not be visible in your remote widget
5/23/2020, 8:37:20 PM Got SIGINT, shutting down Bridge...
5/23/2020, 8:37:20 PM [HarmonyHubWebSocket] INFO - shutdown
5/23/2020, 8:37:24 PM Loaded plugin "homebridge-dummy".
5/23/2020, 8:37:24 PM Loaded plugin "homebridge-fujitsu".
5/23/2020, 8:37:24 PM Loaded plugin "homebridge-harmony".
5/23/2020, 8:37:25 PM Loaded plugin "homebridge-ring".
5/23/2020, 8:37:25 PM Loaded plugin "homebridge-smartthings-v2".
5/23/2020, 8:37:26 PM [Basement TV] HarmonyPlatform Init
5/23/2020, 8:37:26 PM [Basement TV] WARNING - configureAccessory - The platform undefined is not there anymore in your config (name property). It won't be loaded and will be removed from cache.
5/23/2020, 8:37:26 PM [Basement TV] WARNING - configureAccessory - The platform undefined could not be removed
5/23/2020, 8:37:26 PM [Basement TV] DidFinishLaunching
5/23/2020, 8:37:26 PM [Basement TV] (Basement TV)INFO - Loading activities...
5/23/2020, 8:37:26 PM [Basement TV] (Family Room TV)INFO - Loading activities...
5/23/2020, 8:37:26 PM [Basement TV] (Master Bedroom TV)INFO - Loading activities...
5/23/2020, 8:37:26 PM [Basement TV] (Guest Bedroom TV)INFO - Loading activities...
5/23/2020, 8:37:26 PM [Basement TV] INFO - starting hub discovery
5/23/2020, 8:37:32 PM [Basement TV] INFO - stopping hub discovery, hubs found : 10.0.0.108|Basement Hub|16323923,10.0.0.224|Master Bedroom Hub|14359868,10.0.0.208|Family Room Hub|11028932
5/23/2020, 8:37:32 PM [Basement TV] (Basement TV)INFO - received discovered hubs  10.0.0.108|Basement Hub|16323923,10.0.0.224|Master Bedroom Hub|14359868,10.0.0.208|Family Room Hub|11028932
5/23/2020, 8:37:32 PM [Basement TV] (Family Room TV)INFO - received discovered hubs  10.0.0.108|Basement Hub|16323923,10.0.0.224|Master Bedroom Hub|14359868,10.0.0.208|Family Room Hub|11028932
5/23/2020, 8:37:32 PM [Basement TV] (Master Bedroom TV)INFO - received discovered hubs  10.0.0.108|Basement Hub|16323923,10.0.0.224|Master Bedroom Hub|14359868,10.0.0.208|Family Room Hub|11028932
5/23/2020, 8:37:32 PM [Basement TV] (Guest Bedroom TV)INFO - received discovered hubs  10.0.0.108|Basement Hub|16323923,10.0.0.224|Master Bedroom Hub|14359868,10.0.0.208|Family Room Hub|11028932
5/23/2020, 8:37:32 PM [Basement TV] (Guest Bedroom TV)ERROR - hub name was not found Guest Bedroom Hub is expected, found : 10.0.0.108|Basement Hub|16323923,10.0.0.224|Master Bedroom Hub|14359868,10.0.0.208|Family Room Hub|11028932
5/23/2020, 8:37:33 PM [Basement TV] (Master Bedroom TV)INFO - Adding Accessory : Master Bedroom TV-TV
5/23/2020, 8:37:33 PM [Basement TV] (Master Bedroom TV)INFO - configuring Main TV Service
5/23/2020, 8:37:33 PM [Basement TV] (Master Bedroom TV)INFO - Creating TV Service
5/23/2020, 8:37:33 PM [Basement TV] (Master Bedroom TV)INFO - Creating Input Service - Watch Apple TV
5/23/2020, 8:37:33 PM [Basement TV] (Master Bedroom TV)INFO - Creating Input Service - Watch Smart TV
5/23/2020, 8:37:33 PM [Basement TV] (Master Bedroom TV)WARNING - No main Activity that match config file found, default to first one
5/23/2020, 8:37:33 PM [Basement TV] (Master Bedroom TV)INFO - Configuring Main Activity Watch Apple TV
5/23/2020, 8:37:33 PM [Basement TV] (Master Bedroom TV)INFO - Creating TV Speaker Service
5/23/2020, 8:37:33 PM [Basement TV] (Master Bedroom TV)WARNING - setupFoundAccessories - TV accessory added in your bridge, if another plugin is exposing a TV accessory this one might not be visible in your remote widget
5/23/2020, 8:37:33 PM [Basement TV] (Family Room TV)INFO - Adding Accessory : Family Room TV-TV
5/23/2020, 8:37:33 PM [Basement TV] (Family Room TV)INFO - configuring Main TV Service
5/23/2020, 8:37:33 PM [Basement TV] (Family Room TV)INFO - Creating TV Service
5/23/2020, 8:37:33 PM [Basement TV] (Family Room TV)INFO - Creating Input Service - Watch Apple TV
5/23/2020, 8:37:33 PM [Basement TV] (Family Room TV)WARNING - No main Activity that match config file found, default to first one
5/23/2020, 8:37:33 PM [Basement TV] (Family Room TV)INFO - Configuring Main Activity Watch Apple TV
5/23/2020, 8:37:33 PM [Basement TV] (Family Room TV)INFO - Creating TV Speaker Service
5/23/2020, 8:37:33 PM [Basement TV] (Family Room TV)INFO - setupFoundAccessories - TV accessory added as external accessory
5/23/2020, 8:37:33 PM Family Room TV-TV is running on port 39093.
5/23/2020, 8:37:33 PM [Basement TV] (Basement TV)INFO - Adding Accessory : Basement TV-TV
5/23/2020, 8:37:33 PM [Basement TV] (Basement TV)INFO - configuring Main TV Service
5/23/2020, 8:37:33 PM [Basement TV] (Basement TV)INFO - Creating TV Service
5/23/2020, 8:37:33 PM [Basement TV] (Basement TV)INFO - Creating Input Service - Listen to Music
5/23/2020, 8:37:33 PM [Basement TV] (Basement TV)INFO - Configuring Main Activity Watch Apple TV
5/23/2020, 8:37:33 PM [Basement TV] (Basement TV)INFO - Creating TV Speaker Service
5/23/2020, 8:37:33 PM [Basement TV] (Basement TV)INFO - Creating Input Service - Watch Apple TV
5/23/2020, 8:37:33 PM [Basement TV] (Basement TV)INFO - Creating Input Service - Xbox
5/23/2020, 8:37:33 PM [Basement TV] (Basement TV)INFO - setupFoundAccessories - TV accessory added as external accessory
5/23/2020, 8:37:33 PM Basement TV-TV is running on port 43223.

**Comment by askovi** [comment]
According to the logs it cannot discover the Guest Bedroom Hub. Make sure it's connected to the network. Try removing it from the config temporarily to see if that fixes the issue.

**Comment by Corvetteguy19** [comment]
OKay I’ll give that a shot.  On a side note, when posting about configurations, is there a risk in posting the IP addresses of the devices?  I believe those are the LAN address, but I want to be sure I’m not posting anything someone could possibly use to access my house.

**Comment by askovi** [comment]
You can post local ip addresses just fine, those cannot be connected to from the internet. Main things you want to avoid posting are any credentials and access tokens.

**Comment by Corvetteguy19** [comment]
Okay thanks.  So it looks like removing the guest bedroom didn’t help.  I’m going to take down my mesh network to see if that fixes the issue.  Not that that’s a good solution, but I’m curious if I can isolate the problem.

**Comment by Corvetteguy19** [comment]
Alright I might have solved the issue.  This morning I renamed all of my harmony hubs.  I removed the word hub from the back of them.  I’m not sure if this was the issue, but things for the last few hours have been up and running.  It definitely is more stable than it was the other day.  I’ll monitor it and if things stay up for 24 hours I’ll close this.  Thanks guys for the help!

**Comment by Corvetteguy19** [comment]
Hoobs ran for just over 24 hours without problems. Now it’s crashing again every few hours. I’m guessing it’s the harmony plugin.  Now when I restart hoobs it’s re-adding all of the hubs in HomeKit again as new hubs. The original hubs in HomeKit show as no response and I have to remove them. So I’m constantly having to re-update all of my automation. 

Here is what I got for the crash:

5/25/2020, 9:50:41 PM [Basement TV] INFO - starting hub discovery
5/25/2020, 9:50:42 PM Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
    at healthCheck (dgram.js:902:11)
    at Socket.send (dgram.js:627:3)
    at Ping.emit (/home/hoobs/.hoobs/node_modules/harmonyhubjs-discover/lib/ping.js:25:15)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)
5/25/2020, 9:50:42 PM Got SIGTERM, shutting down Bridge...
5/25/2020, 9:50:42 PM [Basement TV] INFO - shutdown
5/25/2020, 9:50:43 PM Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
    at healthCheck (dgram.js:902:11)
    at Socket.send (dgram.js:627:3)
    at Ping.emit (/home/hoobs/.hoobs/node_modules/harmonyhubjs-discover/lib/ping.js:25:15)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)
5/25/2020, 9:50:44 PM Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
    at healthCheck (dgram.js:902:11)
    at Socket.send (dgram.js:627:3)
    at Ping.emit (/home/hoobs/.hoobs/node_modules/harmonyhubjs-discover/lib/ping.js:25:15)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)
5/25/2020, 9:50:45 PM Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
    at healthCheck (dgram.js:902:11)
    at Socket.send (dgram.js:627:3)
    at Ping.emit (/home/hoobs/.hoobs/node_modules/harmonyhubjs-discover/lib/ping.js:25:15)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)

**Comment by Corvetteguy19** [comment]
Any ideas on this?  I’m still having to reboot at least twice a day. Probably more but I’m not monitoring it as much. 

Here is what I’m getting:

5/30/2020, 7:53:09 AM [Basement TV] (Guest Bedroom TV)WARNING - socket closed
5/30/2020, 7:53:19 AM [Basement TV] INFO - starting hub discovery
5/30/2020, 7:53:20 AM Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
    at healthCheck (dgram.js:902:11)
    at Socket.send (dgram.js:627:3)
    at Ping.emit (/home/hoobs/.hoobs/node_modules/harmonyhubjs-discover/lib/ping.js:25:15)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)
5/30/2020, 7:53:20 AM Got SIGTERM, shutting down Bridge...
5/30/2020, 7:53:20 AM [Basement TV] INFO - shutdown
5/30/2020, 7:53:21 AM Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
    at healthCheck (dgram.js:902:11)
    at Socket.send (dgram.js:627:3)
    at Ping.emit (/home/hoobs/.hoobs/node_modules/harmonyhubjs-discover/lib/ping.js:25:15)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)
5/30/2020, 7:53:22 AM Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
    at healthCheck (dgram.js:902:11)
    at Socket.send (dgram.js:627:3)
    at Ping.emit (/home/hoobs/.hoobs/node_modules/harmonyhubjs-discover/lib/ping.js:25:15)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)
5/30/2020, 7:53:23 AM Error [ERR_SOCKET_DGRAM_NOT_RUNNING]: Not running
    at healthCheck (dgram.js:902:11)
    at Socket.send (dgram.js:627:3)
    at Ping.emit (/home/hoobs/.hoobs/node_modules/harmonyhubjs-discover/lib/ping.js:25:15)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)

**Comment by Corvetteguy19** [comment]
Still having the issue. How long is hoobs supposed to stay running for normally?  I would think it would stay up until someone would want to take it down. 

Is there a work around for this?  Meaning if the service stops is there a way to have it automatically restart?

**Comment by Corvetteguy19** [comment]
So I think I might have a better understanding of the issue. I was watching tv the other night and I noticed my harmony hub had randomly disconnected from the network. Given I have four of them, I’m wondering if when they disconnect that’s when I get the WebSocket error. Would this make sense?  If so, does anyone have any ideas why they would randomly disconnect?  I’m using xfinities xfi7 router. I had the same problem with the xfi6.

**Comment by Corvetteguy19** [comment]
So I believe I have solved the issue.  I was using the hub names for the harmony plugin, which were correct, but for some reason it kept causing websocket disconnect issues multiple times a day.  I switched my hubs over to dedicated IP addresses and switched the plugin to use dedicated IP addresses.  Since I’ve done that I have not had issues with Hoobs crashing.  It has remained on for days without issues.  Not sure what was wrong, but something was closing the websocket connection it seems when I was using the hub names.
